src/ga4gh/core/_internal/identifiers.py

```diff
@@ -94,10 +144,25 @@ def ga4gh_identify(vro):
 
     """
     if is_identifiable(vro):
-        digest = ga4gh_digest(vro)
-        pfx = vro.ga4gh.prefix
-        ir = f"{namespace}{curie_sep}{pfx}{ref_sep}{digest}"
+        when_rule = ga4gh_compute_identifier_when.get(GA4GHComputeIdentifierWhen.ALWAYS)
+        do_compute = False
+        ir = None
+        if when_rule == GA4GHComputeIdentifierWhen.ALWAYS:
+            do_compute = True
+        else:
+            ir = getattr(vro, "id", None)
+            if when_rule == GA4GHComputeIdentifierWhen.MISSING:
+                do_compute = ir is None or ir == ""
+            else:  # INVALID
+                do_compute = ir is None or ir == "" or ga4gh_ir_regexp.match(ir) is None
```

Comment:
I think we may be able to do this:

```suggestion
            do_compute = not ir
            if when_rule == GA4GHComputeIdentifierWhen.INVALID:
                do_compute = do_compute or ga4gh_ir_regexp.match(ir) is None
```